ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -265,9 +263,9 @@ addBlockSync
      )
   => ChainDbEnv m blk
   -> BlockToAdd m blk
-  -> m ()
+  -> HapaxProcess m ()
 addBlockSync cdb@CDB {..} BlockToAdd { blockToAdd = b, .. } = do
```

Comment:
Couldn't we lift the whole `do` block?

```suggestion
addBlockSync cdb@CDB {..} BlockToAdd { blockToAdd = b, .. } = lift $ do
```

Idem in other places

Reply:
Done